Arm_Software/Arm_Software.ino

```diff
@@ -0,0 +1,356 @@
+#include "Arm_Software.h"
+
+void setup()
+{
+    Serial.begin(115200);
+    ShoulderTilt.motor.attach(MotorINA_1,MotorINB_1,MotorPWM_1);
+    ShoulderTwist.motor.attach(MotorINA_2,MotorINB_2,MotorPWM_2);
+    ElbowTilt.motor.attach(MotorINA_3,MotorINB_3,MotorPWM_3);
+    ElbowTwist.motor.attach(MotorINA_4,MotorINB_4,MotorPWM_4);
+    Wrist.rightMotor.attach(MotorINA_5,MotorINB_5,MotorPWM_5);
+    Wrist.leftMotor.attach(MotorINA_6,MotorINB_6,MotorPWM_6);
+    Gripper.motor.attach(MotorINA_7,MotorINB_7,MotorPWM_7);
+
+    ShoulderTilt.encoder.attach(Encoder_ShoulderTilt);
+    ShoulderTwist.encoder.attach(Encoder_ShoulderTwist);
+    ElbowTilt.encoder.attach(Encoder_ElbowTilt);
+    ElbowTwist.encoder.attach(Encoder_ElbowTwist);
+    Wrist.tiltEncoder.attach(Encoder_WristTilt);
+    Wrist.twistEncoder.attach(Encoder_WristTwist);
+
+    ShoulderTilt.attachLimitSwitches(LimitSwitchLower_J1, LimitSwitchUpper_J1);
+    ShoulderTwist.attachLimitSwitches(LimitSwitchLower_J2, LimitSwitchUpper_J2);
+    ElbowTilt.attachLimitSwitches(LimitSwitchLower_J3, LimitSwitchUpper_J3);
+
+    ShoulderTilt.setAngleLimits(180,45);
+    ShoulderTwist.setAngleLimits(360,0);
+    ElbowTilt.setAngleLimits(180,30);
+    ElbowTwist.setAngleLimits(360,0);
+
+    ShoulderTilt.encoder.start();
+    ShoulderTwist.encoder.start();
+    ElbowTilt.encoder.start();
+    ElbowTwist.encoder.start();
+    Wrist.tiltEncoder.start();
+    Wrist.twistEncoder.start();
+
+    ElbowTilt.pid.attach( -300.0, 300.0, 27, 0, 0);
+    ElbowTwist.pid.attach( -300.0, 300.0, 15.0, 0, 0 );
+
+    ShoulderTilt.pid.attach( -700.0, 400.0, 20.0, 0, 0 );
+    ShoulderTwist.pid.attach( -600.0, 600.0, 20.0, 0, 0 );
+
+    Wrist.tiltPid.attach( -700.0, 400.0, 20.0, 0, 0);
+    Wrist.twistPid.attach( -600.0, 600.0, 20.0, 0, 0 );
+
+    RoveComm.begin(RC_ARMBOARD_FOURTHOCTET, &TCPServer);
+
+    Watchdog.attach(estop);
+    WatchdogTelemetry.attach(telemetry);
+    Watchdog.start(watchdogTimeout);
+    WatchdogTelemetry.start(ROVECOMM_UPDATE_RATE);
+}
+
+void loop()
+{
+    updatePosition();
+    parsePackets();
+    updatePosition();
```

Comment:
You might also want to only call updatePosition when its needed (e.g. when sending telemetry or calculating closed loop), this will spam the encoder reads as fast as possible and is a little overkill.